evm/ts-scripts/src/updatePeerAddresses.ts

```diff
@@ -0,0 +1,293 @@
+
+const processName = "updatePeerAddresses";
+
+type PeerConfig = {
+  chainId: ChainId;
+  decimals: number;
+  isWormholeRelayingEnabled: boolean;
+  isWormholeEvmChain: boolean;
+  isSpecialRelayingEnabled: boolean;
+
+  managerAddress?: string;
+  transceiverAddress?: string;
+};
+
+init();
+const chains = loadOperatingChains();
+async function run() {
+  // Warning: we assume that the script configuration file is correctly formed
+  const config = (await loadScriptConfig("peers")) as PeerConfig[];
+  console.log(`Start ${processName}!`);
+
+  const results = await Promise.all(
+    chains.map(async (chain) => {
+      let result: {
+        chainId: ChainId;
+        peerUpdateTxs: string[];
+        transceiverUpdateTxs: string[];
+        error?: unknown;
+      };
+      try {
+        result = await registerPeers(chain, config);
+      } catch (error: unknown) {
+        return { chainId: chain.chainId, peerUpdateTxs: [] as string[], error };
+      }
+
+      return result;
+    })
+  );
+
+  for (const result of results) {
+    if ("error" in result) {
+      console.error(
+        `Error configuring contract for chain ${result.chainId}: ${inspect(
+          result.error
+        )}`
+      );
+      continue;
+    }
+
+    console.log(
+      `NttManager set peer txs for chain ${
+        result.chainId
+      }: \n  ${result.peerUpdateTxs.join("\n  ")}`
+    );
+
+    console.log(
+      `NttManager set transceiver peer txs for chain ${
+        result.chainId
+      }: \n  ${result.transceiverUpdateTxs.join("\n  ")}`
+    );
+  }
+}
+
+async function registerPeers(
+  chain: ChainInfo,
+  peers: PeerConfig[]
+): Promise<{
+  chainId: ChainId;
+  peerUpdateTxs: string[];
+  transceiverUpdateTxs: string[];
+  error?: unknown;
+}> {
+  const log = (...args: any[]) => console.log(`[${chain.chainId}]`, ...args);
+
+  const managerContract = await getManagerContract(chain);
+  const transceiverContract = await getTransceiverContract(chain);
+
+  const peerUpdateTxs: string[] = [];
+  const transceiverUpdateTxs: string[] = [];
+  const managerConfig = await getChainConfig<ManagerConfig>(
+    "managers",
+    chain.chainId
+  );
+  for (const peer of peers) {
+    if (peer.chainId === chain.chainId) continue;
+
+    const config = await getChainConfig<PeerConfig>("peers", peer.chainId);
+
+    if (!config.decimals)
+      return {
+        chainId: chain.chainId,
+        peerUpdateTxs,
+        transceiverUpdateTxs,
+        error: "No 'decimals' configuration found",
+      };
+
+    const desiredPeerAddress = await getNormalizedPeerManagerAddress(
+      peer,
+      chain
+    );
+
+    if (!desiredPeerAddress)
+      return {
+        chainId: chain.chainId,
+        peerUpdateTxs,
+        transceiverUpdateTxs,
+        error: "No 'managerAddress' found",
+      };
+
+    const inboundLimit = managerConfig.inboundLimit.find(
+      (x) => x.chainId === peer.chainId
+    )?.limit;
+
+    if (!inboundLimit) {
```

Comment:
```suggestion
    if (inboundLimit === undefined) {
```

If the rate limiter is off, this can be 0 (right?)

Reply:
thanks!